### PR TITLE
0.12 Bevy Asset V2

### PR DESCRIPTION
**Pull request by cart**
Part of #754

### CODE REVIEW

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+* **Compression**: Minimize the disk space and/or bandwidth that an asset takes up in deployed apps
+* **Transformation**: Some "asset source files" aren't in the right format by default. You can have an asset of type `A` and transform it into type `B`.
+
+If you are building an app that tests the limits of your hardware with optimal formats ... or you just want to cut down on startup / loading times, asset preprocessing is for you.
+
+For an in-depth technical breakdown of the implementation we chose, check out the [Bevy Asset V2 pull request](https://github.com/bevyengine/bevy/pull/8624).
+
+### Enabling Processing
+
+To enable processing, just configure your [`AssetPlugin`] like this:
```
Comment (alice-i-cecile) [inline content/news/2023-10-21-bevy-0.12/index.md:49]:
```suggestion
To enable asset pre-processing, just configure your [`AssetPlugin`] like this:
```

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+* **Compression**: Minimize the disk space and/or bandwidth that an asset takes up in deployed apps
+* **Transformation**: Some "asset source files" aren't in the right format by default. You can have an asset of type `A` and transform it into type `B`.
+
+If you are building an app that tests the limits of your hardware with optimal formats ... or you just want to cut down on startup / loading times, asset preprocessing is for you.
+
+For an in-depth technical breakdown of the implementation we chose, check out the [Bevy Asset V2 pull request](https://github.com/bevyengine/bevy/pull/8624).
+
+### Enabling Processing
```
Comment (alice-i-cecile) [inline content/news/2023-10-21-bevy-0.12/index.md:47]:
```suggestion
### Enabling Pre-Processing
```
I found this very hard to parse intially.

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 3]
```diff
@@ -15,9 +15,351 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+The [`AssetProcessor`] can close (either intentionally or unintentionally) at any point in time and it will pick up right where it left off!
+
+If a Bevy App asks to load an asset that is currently being processed (or re-processed), the load will asynchronously wait until both the processed asset and its meta file have been written. This ensures that a loaded asset file and meta file always "match" for a given load.
+
+### Asset Meta Files
+
+Assets now support (optional) `.meta` files. This enables configuring things like:
+
+* **The asset "action"**
```
Comment (alice-i-cecile) [inline content/news/2023-10-21-bevy-0.12/index.md:93]: This explanation could be clearer. It's hard to understand what this might be before reading the examples, and even then it requires some logical jumps.

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 4]
```diff
@@ -15,9 +15,351 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+[`Process`]: https://dev-docs.bevyengine.org/bevy/asset/processor/trait.Process.html
+[`ImageLoader`]: https://dev-docs.bevyengine.org/bevy/render/texture/struct.ImageLoader.html
+
+### `CompressedImageSaver`
+
+![processed sponza](processed_sponza.png)
+<div style="font-size: 1.0rem" class="release-feature-authors">Sponza scene with textures processed into Basis Universal (with mipmaps) using Bevy Asset V2</div>
+
+**Bevy 0.12** ships with a barebones [`CompressedImageSaver`] that writes images to [Basis Universal](https://github.com/BinomialLLC/basis_universal) (a GPU-friendly image interchange format) and generates mipmaps. This can be enabled with the `basis-universal` cargo feature.
```
Comment (alice-i-cecile) [inline content/news/2023-10-21-bevy-0.12/index.md:173]: Mipmaps would really benefit from a small explanation or link. Ideally both!

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 5]
```diff
@@ -15,9 +15,351 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+let path = handle.path();
+```
+
+Handles now also use a smaller / cheaper-to-look-up [`AssetIndex`] internally, which uses generational indices to look up assets in dense storage.
+
+[`AssetPath`]: https://dev-docs.bevyengine.org/bevy/asset/struct.AssetPath.html
+[`Handle`]: https://dev-docs.bevyengine.org/bevy/asset/enum.Handle.html
+[`AssetIndex`]: https://dev-docs.bevyengine.org/bevy/asset/struct.AssetIndex.html
+
+### True Copy on Write Asset Paths
```
Comment (alice-i-cecile) [inline content/news/2023-10-21-bevy-0.12/index.md:350]:
```suggestion
### True Copy-on-Write Asset Paths
```

File: content/news/2023-10-21-bevy-0.12/index.md (hunk 3 above)
Comment (cart) [reply content/news/2023-10-21-bevy-0.12/index.md:93]: Good call!
